pkg/ironicneutronagent/volumes.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2023 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ironicneutronagent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GetVolumes -
+func GetVolumes(name string) []corev1.Volume {
+	var scriptsVolumeDefaultMode int32 = 0755
+	var config0640AccessMode int32 = 0640
+
+	return []corev1.Volume{
+		{
+			Name: "scripts",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					DefaultMode: &scriptsVolumeDefaultMode,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: name + "-scripts",
+					},
+				},
+			},
+		},
+		{
+			Name: "config-data",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					DefaultMode: &config0640AccessMode,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: name + "-config-data",
```

Comment:
Just thinking outloud, since we're trying to carry the tripleo config modeling across, shouldnt' we look at dropping it and keeping it confined to containers using env vars coupled with the container launch?

Reply:
You are thinking about using oslo's Environment source[1]?
It is an interesting idea, but that would have to be an agreed change with the larger openstack-k8s-operators community.


[1] https://opendev.org/openstack/oslo.config/src/branch/master/oslo_config/sources/_environment.py